### DISCUSSION TITLE
Sorting `suspensions` in `TestClock.run`

### DISCUSSION THREAD

**Discussion by danieleformichelli** [pointfreeco/swift-clocks#7] [Q&A] [answered]
Hello, I have a question about the `TestClock` 👋 

As [we are adding suspension without sorting](https://github.com/pointfreeco/swift-clocks/blob/d8ff0c92e8c9d139356adae1c3c079b12e81d4df/Sources/Clocks/TestClock.swift#L112) them [until `advance` is called](https://github.com/pointfreeco/swift-clocks/blob/d8ff0c92e8c9d139356adae1c3c079b12e81d4df/Sources/Clocks/TestClock.swift#L172), shouldn't the `run` method sort them as well [when getting the next suspension](https://github.com/pointfreeco/swift-clocks/blob/d8ff0c92e8c9d139356adae1c3c079b12e81d4df/Sources/Clocks/TestClock.swift#L241), in case it is invoked without calling `advance` first??

If yes, wouldn't be simpler and less error-prone to just use insertion sort when adding suspensions to the array?

Thanks a lot! 🙏

**Comment by stephencelis** [comment, marked as answer]
@danyf90 Chatted about this with @mbrandonw and we totally agree. Would you be up for PRing that functionality and, if you wanna go above and beyond, capture things in a test? Any help is appreciated, but we're also happy to take on any of the work if you don't have time to dive in :smile:

**Reply by danieleformichelli** [reply]
Sure! I'll try to open a PR in the following days 🚀

**Reply by danieleformichelli** [reply]
Opened a [PR here](https://github.com/pointfreeco/swift-clocks/pull/8), in the end the behavior was probably fine, but I still think the approach in the PR is cleaner and safer, so I have opened it anyway
